tests/smoke_test.go

```diff
@@ -0,0 +1,131 @@
+package tests
+
+import (
+	"errors"
+	"fmt"
+	"github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/apis/sparkoperator.k8s.io/v1beta2"
+	"github.com/mesosphere/kudo-spark-operator/tests/utils"
+	log "github.com/sirupsen/logrus"
+	v12 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestShuffleApp(t *testing.T) {
+	spark := utils.SparkOperatorInstallation{}
+	err := spark.InstallSparkOperator()
+	defer spark.CleanUp()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	const expectedExecutorCount = 2
+	const expectedGroupCount = 12000
+
+	jobName := "shuffle-app"
+	job := utils.SparkJob{
+		Name:     jobName,
+		Template: "spark-shuffle-job.yaml",
+		Params: map[string]interface{}{
+			"executor_count": expectedExecutorCount,
+			"group_count":    fmt.Sprintf("\"%d\"", expectedGroupCount),
+		},
+	}
+
+	err = spark.SubmitJob(&job)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = spark.WaitForJobState(job, v1beta2.RunningState, 5*time.Minute)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Wait for correct number of executors to show up
+	err = utils.Retry(30*time.Second, time.Second, func() error {
+		executors, err := spark.JobExecutors(job)
+		if err != nil {
+			return err
+		} else if len(executors) != expectedExecutorCount {
+			return errors.New(fmt.Sprintf("The number of executors is %d, but %d is expected", len(executors), expectedExecutorCount))
+		}
+		return nil
+	})
+	if err != nil {
+		t.Error(err)
+	}
+
+	spark.WaitForOutput(job, fmt.Sprintf("Groups count: %d", expectedGroupCount))
+}
+
+func TestMockTaskRunner(t *testing.T) {
+	spark := utils.SparkOperatorInstallation{}
+	err := spark.InstallSparkOperator()
+	defer spark.CleanUp()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	jobName := "mock-task-runner"
+	job := utils.SparkJob{
+		Name:     jobName,
+		Template: "spark-mock-task-runner-job.yaml",
+	}
+	expectedExecutorCount := 1
+
+	// Submit the job and wait for it to start
+	err = spark.SubmitJob(&job)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = spark.WaitForJobState(job, v1beta2.RunningState, 5*time.Minute)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Wait for correct number of executors to show up
+	err = utils.Retry(30*time.Second, time.Second, func() error {
```

Comment:
same comment here regarding extracting timeouts to constants or config structure

Reply:
done